### PR TITLE
feat:  automatically add users to team during synchronization

### LINKED ISSUE #1601: [FEATURE]: Auto-add users to pre-defined team
## Summary

Automatically add users to pre-defined team (if set in env vars) during AD-App users synchronization

### PR DESCRIPTION
**Pull request by joaofrparreira**
Relates to:
- #1601 

## Proposed Changes

  - Automatically add's new users to a team defined in env vars during AD-App users synchronization cron job
  - Changes insertion of users from each one to a user batch

This pull request closes #1601

### CODE REVIEW

File: backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts [hunk 1]
```diff
@@ -43,6 +66,8 @@ export class SynchronizeADUsersCronUseCase implements SynchronizeADUsersCronUseC
 				throw new Error('Split app users list is empty.');
 			}
 
+			const usersAppFiltered = usersApp.filter((user) => user.email.endsWith('@xgeeks.com'));
```
Comment (nunocaseiro) [inline backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts:69]: What do you think if we add a new env var for the team email's domain? e.g. "@xgeeks.com"

You've added an env var to define the team name so maybe it also makes sense to add the email domain

File: backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts [hunk 2]
```diff
@@ -91,28 +116,51 @@ export class SynchronizeADUsersCronUseCase implements SynchronizeADUsersCronUseC
...
-			} catch (err) {
-				this.logger.error(
-					`An error as occurred while creating user with email '${user.mail}' through the syncronize AD Users Cron. Message: ${err.message}`
-				);
+				};
+			});
+
+			const createdUsers = await this.createUserService.createMany(usersToCreate);
+
+			if (typeof team !== 'undefined') {
```
Comment (nunocaseiro) [inline backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts:145]:
```suggestion
			if (team) {
```

Do we really need to check the type of the team?

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts (hunk 1 above)
Comment (joaofrparreira) [reply backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts:69]: It makes all the sense

File: backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts (hunk 2 above)
Comment (joaofrparreira) [reply backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts:145]: Force of habit, in this case bot are equally fine to use.

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Pull request merged**